src/main/scala/UpdateXmlGenerator.scala

```diff
@@ -0,0 +1,91 @@
+import java.time.{Instant, LocalDateTime, ZoneOffset, ZonedDateTime}
+
+import com.gu.contentatom.thrift.{Atom, AtomData}
+import com.gu.contentatom.thrift.atom.media.{Asset, Platform}
+
+import scala.xml.{Elem, NodeSeq}
+
+object UpdateXmlGenerator {
+  private def fieldOptionIterable[T](fieldName:String, atomField:Option[Iterable[T]])=atomField.map({ value=>
+    <field><name>{fieldName}</name>{value.map({ entry=> <value>{entry}</value>})}</field>
+  })
+
+  private def fieldOption[T](fieldName:String, atomField:Option[T])=atomField.map({ value=>
+    <field><name>{fieldName}</name> <value>{value}</value></field>
+  })
+
+  def asIsoTimeString(epochTime:Long):String = ZonedDateTime.ofInstant(Instant.ofEpochSecond(epochTime),ZoneOffset.UTC).toString
+
+  private def youtubePortion(atom:Atom, currentTime: LocalDateTime):Option[Elem] = {
+    val mediaContent = atom.data.asInstanceOf[AtomData.Media].media
+
+    val ytAssets=mediaContent.assets.filter(_.platform==Platform.Youtube)
+
+    //logger.debug(s"ytAssets: $ytAssets")
+    if(ytAssets.isEmpty) return Some(
+      <list><field><name>gnm_master_youtube_status</name> <value>Unpublished</value></field></list>
+    )
+
+    Some(
+      // the containing <list> is not used by Vidispine, but is used here so that the scala compiler has a single <elem> to return
+    <list>
+      <field><name>gnm_master_youtube_title</name> <value>{mediaContent.title}</value></field>
+      {fieldOption("gnm_master_youtube_description",mediaContent.description).getOrElse("")}
+      {fieldOptionIterable("gnm_master_youtube_keywords",mediaContent.metadata.flatMap(_.tags)).getOrElse("")}
+      {fieldOption("gnm_master_youtube_category", mediaContent.metadata.flatMap(_.categoryId)).getOrElse("")}
+      <field><name>gnm_master_youtube_allowcomments</name> <value>{mediaContent.metadata.flatMap(_.commentsEnabled.map(if(_) "allow_comments" else "")).getOrElse("")}</value></field>
+      <field><name>gnm_master_youtube_uploadstatus</name> <value>Upload Succeeded</value></field>
+      {fieldOption("gnm_master_youtube_remove",mediaContent.metadata.flatMap(_.expiryDate).map(asIsoTimeString(_))).getOrElse()}
+      <field><name>gnm_master_youtube_holdingimage_16x9</name> <value/></field>
+      {fieldOption("gnm_master_youtube_channelid",mediaContent.metadata.flatMap(_.channelId)).getOrElse("")}
+      {fieldOption("gnm_master_youtube_license",mediaContent.metadata.flatMap(_.license)).getOrElse("")}
+      <field><name>gnm_master_youtube_status</name> <value>Published</value></field>
+      <field><name>gnm_master_youtube_holdingimage</name><value/></field>
+      <field><name>gnm_master_youtube_youtubeurl</name>{ytAssets.flatMap({ asset:Asset => <value>https://www.youtube.com/watch?v={asset.id}</value>})}</field>
+      <field><name>gnm_master_youtube_publish</name> <value>{currentTime.toString}</value></field>
+    </list>
+    )
+  }
+
+  def makeContentXml(atom:Atom, currentTime: LocalDateTime):Elem = {
```

Comment:
I think we can make the type of the atom param AtomData.Media as we know its that type [here](https://github.com/guardian/multimedia-launchdetector-v3/pull/2/files#diff-0f7307c57e8188e6823e612cad21c1dcR87)

Reply:
can't do that here as I am getting the content change details and the atom root title/id